From 6389b257ea72e3bd1493ae8d174b4a28dfa25ba6 Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Wed, 16 Jul 2025 14:46:06 +0000
Subject: [PATCH 31/31] Enable all codec2 components except c2.android when
 debug.stagefright.ccodec=0

`c2.android` only works on graphics stack with proper YUV support and does not work on Waydroid at all

Change-Id: If722001d129ab7c58d3a5be737ffd61a3eb26005
Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 media/codec2/sfplugin/Codec2InfoBuilder.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/media/codec2/sfplugin/Codec2InfoBuilder.cpp b/media/codec2/sfplugin/Codec2InfoBuilder.cpp
index c0b8065500..0f9765a7e2 100644
--- a/media/codec2/sfplugin/Codec2InfoBuilder.cpp
+++ b/media/codec2/sfplugin/Codec2InfoBuilder.cpp
@@ -557,7 +557,8 @@ status_t Codec2InfoBuilder::buildMediaCodecList(MediaCodecListWriter* writer) {
             // TODO: Remove this block once all codecs are enabled by default.
             switch (option) {
             case 0:
-                continue;
+                if (hasPrefix(canonName, "c2.android")) continue;
+                break;
             case 1:
                 if (hasPrefix(canonName, "c2.vda.")) {
                     break;
-- 
2.50.1

